core/devops.go

```diff
@@ -209,7 +210,18 @@ func (d *Devops) invokeOrQuery(ctx context.Context, chaincodeInvocationSpec *pb.
 	}
 
 	// Now create the Transactions message and send to Peer.
-	uuid := util.GenerateUUID()
+	var customUUIDgenAlg = strings.ToLower(chaincodeInvocationSpec.UuidGenerationAlg)
+	var uuid string
+	var generr error
+	if customUUIDgenAlg != "" {
+		uuid, generr = util.GenerateUUIDWithAlg(customUUIDgenAlg, chaincodeInvocationSpec.ChaincodeSpec.CtorMsg.Args[0])
+		if generr != nil {
+			return nil, generr
+		}
+	} else {
+		uuid = util.GenerateUUID()
+	}
+	devopsLogger.Info("Transaction ID: %v", uuid)
```

Comment:
@tamasblummer @gabre just noticed on line 217 we are using just Args[0]... is that the intent ? Doesn't the uuid need to be computed on the whole content (and even the whole transaction bytes) ?